sound/soc/intel/boards/cml_rt1011_rt5682.c

```diff
@@ -547,8 +547,8 @@ static int snd_cml_rt1011_probe(struct platform_device *pdev)
 	if (sof_rt1011_quirk & (SOF_RT1011_SPEAKER_TL |
 				SOF_RT1011_SPEAKER_TR)) {
 		for_each_card_prelinks(&snd_soc_card_cml, i, dai_link) {
-			if (!strcmp(cml_rt1011_rt5682_dailink[i].codecs[0].dai_name,
-				    CML_RT1011_CODEC_DAI)) {
+			if (!strncmp(cml_rt1011_rt5682_dailink[i].codecs[0].dai_name,
+				     CML_RT1011_CODEC_DAI, sizeof(CML_RT1011_CODEC_DAI))) {
```

Comment:
@fredoh9 why should we change this? What do you mean by safer? Im curious about what issue this prevents?

Reply:
@fredoh9 I agree with @ranj063 - when comparing to a constant string the above change makes no sense - you already know with certainty, that that's the length of the string.